components/brave_new_tab_ui/reducers/new_tab_reducer.ts

```diff
@@ -56,19 +57,9 @@ export const newTabReducer: Reducer<NewTab.State | undefined> = (state: NewTab.S
       }
 
       if (initialDataPayload.wallpaperData) {
-        // Payload passed from native UI doesn't have 'type' property. We should
-        // fix up here.
```

Comment:
This is what I was mistaken about. By implementing `NTPBackgroundImageService`, we can get proper initial wallpaper data.